From 946b49ebcbf47e030a6e00ce1a699f3b055f638a Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Date: Wed, 17 Mar 2010 17:37:05 -0500
Subject: [PATCH 16/37] Catch errors in recursive relink targets

If make relink fails in a subdirectory, we need to catch the error
otherwise make will continue iterating the 'for' loop.

Signed-off-by: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Reviewed-by: Dan Nicholson <dbn.lists@gmail.com>
---
 hw/Makefile.am        |    2 +-
 hw/kdrive/Makefile.am |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git xorg-server-1.7.99.902/hw/Makefile.am xorg-server-1.7.99.902/hw/Makefile.am
index 92b007a..6c2cc6b 100644
--- xorg-server-1.7.99.902/hw/Makefile.am
+++ xorg-server-1.7.99.902/hw/Makefile.am
@@ -38,4 +38,4 @@ SUBDIRS =			\
 DIST_SUBDIRS = dmx xfree86 vfb xnest xwin xquartz kdrive
 
 relink:
-	$(AM_V_at)for i in $(SUBDIRS) ; do $(MAKE) -C $$i relink ; done
+	$(AM_V_at)for i in $(SUBDIRS) ; do $(MAKE) -C $$i relink || exit 1 ; done
diff --git xorg-server-1.7.99.902/hw/kdrive/Makefile.am xorg-server-1.7.99.902/hw/kdrive/Makefile.am
index 55d940f..471ca89 100644
--- xorg-server-1.7.99.902/hw/kdrive/Makefile.am
+++ xorg-server-1.7.99.902/hw/kdrive/Makefile.am
@@ -27,4 +27,4 @@ SUBDIRS =			\
 DIST_SUBDIRS = fbdev ephyr src linux fake
 
 relink:
-	$(AM_V_at)for i in $(SERVER_SUBDIRS) ; do $(MAKE) -C $$i relink ; done
+	$(AM_V_at)for i in $(SERVER_SUBDIRS) ; do $(MAKE) -C $$i relink || exit 1 ; done
-- 
1.7.0.2

